### DISCUSSION TITLE
Classname in livewire-components.php

### DISCUSSION THREAD

**Discussion by Fred7877** [livewire/livewire#3016] [7. Bugs (archived)]
### Description
When modifying the class_namespace in the configuration file with the value 'App.modules.directory.livewire.hierarchy-user', for example, the array key looks like this: app.modules.directory.livewire.hierarchy-user, which is not practical.
When modifying the class_namespace in the configuration file with the value 'App.modules.directory.livewire.hierarchy-user', for example, the array key looks like this: app.modules.directory.livewire.hierarchy-user, which is not practical.

### Exact steps to reproduce
Create a subdirectory in Http (Modules for example) and create a Livewire subdirectory
Execute the command php artisan livewire:discover and look into  bootstrap\cache\livewire-components.php file

### Stripped-down, copy-pastable code snippets
I would like to propose a modification in the LivewireComponentsFinder.php file in the build() method and change the return in this way return [Str::afterLast($class::getName(), '.') => $class];

**Context**
- Livewire version: [2.4.3]
- Laravel version: [ 8.36.2]

**Comment by calebporzio** [comment]
Sorry, I don't understand

**Comment by Fred7877** [comment]
Sorry, my english is to bad, I start again.

**Description**
When you change the class_namespace key in the configuration file to 'App\\Modules\\*\\Livewire', for example, the array key in the livewire-components.php file looks like this: app.modules.directory.livewire.hierarchy-user, which is not practical.

**Exact steps to reproduce**
Create a subdirectory Modules  in App and create a Livewire subdirectory inside.
Change class_namespace key in config/livewire.php with App\\Modules\\*\\Livewire. (* for any directory)
Execute the command php artisan livewire:discover and look into bootstrap\cache\livewire-components.php file
The key in the table that represents the component name will be of the form "app.modules.directory.livewire.hierarchy-user" instead of simply "hierarchy-user"

**Stripped-down, copy-pastable code snippets**
For fix that, maybe i would like to propose a modification in the LivewireComponentsFinder.php file in the build() method and change the return in this way return [Str::afterLast($class::getName(), '.') => $class];

I hope it is clearer :)

**Comment by calebporzio** [comment]
Interesting, thanks for following up. That does sound like a bug.

Do you feel about to open a pill request with at least a failing test?
